DragaliaAPI/Features/Missions/MissionInitialProgressionService.cs

```diff
@@ -14,81 +15,104 @@
-                    .Select(x => x.Level)
-                    .FirstOrDefaultAsync(),
-            MissionProgressType.FortLevelup
-                => await this.fortRepository.Builds
-                    .Where(x => x.PlantId != FortPlants.TheHalidom)
-                    .SumAsync(
-                        x => x.BuildEndDate == DateTimeOffset.UnixEpoch ? x.Level : x.Level - 1
-                    ),
-            MissionProgressType.QuestCleared => await GetQuestClearedCount(requirement),
-            MissionProgressType.CharacterBuildup => await GetCharacterBuildupCount(requirement),
-            MissionProgressType.WyrmprintAugmentBuildup
-                => await GetWyrmprintBuildupCount(requirement),
-            MissionProgressType.WeaponEarned => await GetWeaponEarnedCount(requirement),
-            MissionProgressType.WeaponRefined => await GetWeaponRefinedCount(requirement),
-            MissionProgressType.VoidBattleCleared => await GetVoidBattleClearedCount(requirement),
-            MissionProgressType.ItemSummon => 0, // TODO: As daily quests also use this, should we make it actually count the item summons?
+            MissionCompleteType.FortPlantBuilt
+            or MissionCompleteType.FortPlantPlaced
+                => await fortMissionProgressionService.GetFortPlantCount(
+                    (FortPlants)progressionInfo.Parameter!
+                ),
+            MissionCompleteType.FortLevelUp
+                => await fortMissionProgressionService.GetTotalFortLevel(),
+            MissionCompleteType.QuestCleared => await GetQuestClearedCount(progressionInfo),
+            MissionCompleteType.QuestStoryCleared
+                => await storyRepository.QuestStories.AnyAsync(
+                    x => x.StoryId == progressionInfo.Parameter && x.State == StoryState.Read
+                )
+                    ? 1
+                    : 0,
+            MissionCompleteType.EventGroupCleared
+                => await GetQuestGroupClearedCount(progressionInfo),
+            MissionCompleteType.WeaponEarned => await GetWeaponEarnedCount(progressionInfo),
+            MissionCompleteType.WeaponRefined => await GetWeaponRefinedCount(progressionInfo),
+            MissionCompleteType.AbilityCrestBuildupPlusCount
+                => await GetWyrmprintBuildupCount(progressionInfo),
+            MissionCompleteType.CharacterBuildupPlusCount
+                => await GetCharacterBuildupCount(progressionInfo),
+            MissionCompleteType.PlayerLevelUp
+                => (await userDataRepository.GetUserDataAsync()).Level,
+            MissionCompleteType.AbilityCrestTotalPlusCountUp
+                => (
+                    await abilityCrestRepository.AbilityCrests
+                        .Where(
+                            x =>
+                                progressionInfo.Parameter == null
+                                || x.AbilityCrestId == (AbilityCrests)progressionInfo.Parameter
```

Comment:
is this the "upgrade a wyrmprint with a total of X augments"? not sure I follow why the parameter is a crest

either way, may be worth extracting this query into a func

Reply:
this argument is not used atm (in the currently implemented missions) I just added it if we ever need to implement missions targeting a specific wyrmprint, also it brings it in line with the other chara/dragon methods that also contain the specific id